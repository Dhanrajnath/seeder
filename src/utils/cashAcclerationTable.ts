export interface ITableCellObject {
    value: string;
    percentage?: string;
    discount?: string;
};

export interface IMyContractsTableDataDetail {
    Name: string;
    Type: string;
    perPayment: string;
    termLength: ITableCellObject;
    paymentAmount: ITableCellObject;
};

export interface IMyCashkicksTableDataDetail {
    Name: string;
    Status: "Pending" | "Completed";
    Maturity: string;
    totalRecieved: ITableCellObject;
    totalFinanced: string;
};

export interface ITableColumnLabels {
    label: string;
    column: string
};

export const myContractsTableColumnLabels: ITableColumnLabels[] = [
    { label: 'Name', column: 'Name' },
    { label: 'Type', column: 'Type' },
    { label: 'Per payment', column: 'perPayment' },
    { label: 'Term length', column: 'termLength' },
    { label: 'Payment amount', column: 'paymentAmount' }
];

export const myCashkicksTableColumnLabels: ITableColumnLabels[] = [
    { label: 'Name', column: 'Name' },
    { label: 'Status', column: 'Status' },
    { label: 'Maturity', column: 'Maturity' },
    { label: 'Total recieved', column: 'totalRecieved' },
    { label: 'Total financed', column: 'totalFinanced' }
];

export const myContractsTableData: IMyContractsTableDataDetail[] = [
    {
        Name: 'Contracts1',
        Type: 'Monthly',
        perPayment: '$12,000.25',
        termLength: { value: '12 months', percentage: '12.0% fee' },
        paymentAmount: { value: '$126,722.64' }
    },
    {
        Name: 'Contracts2',
        Type: 'Monthly',
        perPayment: '$6,000.00',
        termLength: { value: '12 months', percentage: '12.0% fee' },
        paymentAmount: { value: '$23,277.36', discount: '$63,380.00' },
    }
];

export const myCashkicksTableData: IMyCashkicksTableDataDetail[] = [
    {
        Name: 'My first advance',
        Status: "Pending",
        Maturity: "Apr 03, 2022",
        totalRecieved: { value: "$150,000.00", percentage: "12% fee" },
        totalFinanced: "$170,454,55"
    }
];