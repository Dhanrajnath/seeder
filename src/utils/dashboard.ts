import CalendarSvg from "../../public/assets/calendar.svg";
import DocumentDownloadSvg from "../../public/assets/document-download.svg";
import PercentageSquareSvg from "../../public/assets/percentage-square.svg";

export const dashboardCardItems = [
    {
        icon: CalendarSvg,
        infoText: 'Term cap',
        title: '12 months'
    },
    {
        icon: DocumentDownloadSvg,
        infoText: 'Available credit',
        title: '$709.45k'
    },
    {
        icon: PercentageSquareSvg,
        infoText: 'Max interest rate',
        title: '12.00%'
    }
];