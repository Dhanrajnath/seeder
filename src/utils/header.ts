export const headerData = {
    title: 'Cash accleration',
    description: 'Place to create new cash kicks to run business'
}