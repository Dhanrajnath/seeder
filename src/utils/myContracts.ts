export interface IMyContractsTableDataDetail {
    Name: string;
    Type: string;
    perPayment: string;
    termLength: string;
    termLengthPercentage: string;
    paymentAmount: string;
    paymentAmountDiscount?: string;
};

export interface IMyContractsTableColumnLabels {
    label: string;
    column: string
}

export const myContractsTableColumnLabels: IMyContractsTableColumnLabels[] = [
    { label: 'Name', column: 'Name' },
    { label: 'Type', column: 'Type' },
    { label: 'Per payment', column: 'perPayment' },
    { label: 'Term length', column: 'termLength' },
    { label: 'Payment amount', column: 'paymentAmount' }
];

export const myContractsTableData: IMyContractsTableDataDetail[] = [
    {
        Name: 'Contracts1',
        Type: 'Monthly',
        perPayment: '$12,000.25',
        termLength: '12 months',
        termLengthPercentage: '12.0% fee',
        paymentAmount: '$126,722.64'
    },
    {
        Name: 'Contracts2',
        Type: 'Monthly',
        perPayment: '$6,000.00',
        termLength: '12 months',
        termLengthPercentage: '12.0% fee',
        paymentAmount: '$23,277.36',
        paymentAmountDiscount: '$63,380.00'
    }
];