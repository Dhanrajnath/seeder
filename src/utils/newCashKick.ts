export const newCashkick = {
    title: "Launch a new Cash Kick",
    description: "You have upto $ available for a new cash advance",
    availableCashKick: "709,546.00",
    buttonName: "New Cash Kick"
};