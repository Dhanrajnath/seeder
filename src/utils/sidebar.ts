import HomeInActiveSvg from "../../public/assets/home-inactive.svg";
import CoinInActiveSvg from "../../public/assets/coin-inactive.svg";
import CoinActiveSvg from "../../public/assets/coin-active.svg";
import HomeActiveSvg from "../../public/assets/home-active.svg";

export type SidebarItemType = {
    id: number;
    activeIconPath: string;
    inActiveIconPath: string;
    itemName: string;
};

export const sidebarItems: SidebarItemType[] = [
    {
        id: 1,
        activeIconPath: HomeActiveSvg,
        inActiveIconPath: HomeInActiveSvg,
        itemName: 'Home'
    },
    {
        id: 2,
        activeIconPath: CoinActiveSvg,
        inActiveIconPath: CoinInActiveSvg,
        itemName: 'Cash Accleration'
    }
];

export const SEEDER = 'Seeder';
export const WATCH_HOW_TO = 'Watch how to';