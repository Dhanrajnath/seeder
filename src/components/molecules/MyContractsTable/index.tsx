import { Grid, Table as MUITable, TableBody, TableCell, TableHead, TableRow } from "@mui/material";
import theme from "../../../utils/theme";
import Typography from "../../atoms/Typography";
import { ITableColumnLabels, IMyContractsTableDataDetail, IMyCashkicksTableDataDetail } from "../../../utils/cashAcclerationTable";

export interface ITableProps {
    tableColumns: ITableColumnLabels[];
    tableData: IMyContractsTableDataDetail[] | IMyCashkicksTableDataDetail[];
};

const Table = ({ tableColumns, tableData }: ITableProps) => {
    return (
        <MUITable
            sx={{
                width: theme.spacing(249),
                maxHeight: theme.spacing(27),
                background: theme.palette.common.black
            }}
        >
            <TableHead sx={{ height: theme.spacing(11) }}>
                <TableRow>
                    {tableColumns.map((column) => (
                        <TableCell
                            variant="head"
                            align="left"
                            width="20%"
                            sx={{
                                pl: theme.spacing(9),
                            }}
                            key={column.column}
                        >
                            <Typography
                                variant="body2"
                                sx={{ color: theme.palette.text.disabled }}
                            >
                                {column.label}
                            </Typography>
                        </TableCell>
                    ))}
                </TableRow>
            </TableHead>
            <TableBody>
                {tableData.map((element, index) => (
                    <TableRow key={index} sx={{ height: theme.spacing(15) }}>
                        {Object.values(element).map((element, i) => (
                            <TableCell
                                key={i}
                                variant="body"
                                align="left"
                                sx={{
                                    pl: theme.spacing(9),
                                }}
                            >
                                <Grid container>
                                    <Grid item>
                                        {typeof element === 'object' ?
                                            <>
                                                <Typography
                                                    variant="body2"
                                                    color={
                                                        i === 0 ?
                                                            theme.palette.text.primary
                                                            : theme.palette.text.disabled}
                                                >
                                                    {element?.value}
                                                </Typography>
                                                {element?.percentage ?
                                                    <Typography
                                                        variant="caption"
                                                        color={theme.palette.text.disabled}
                                                    >
                                                        {element?.percentage}
                                                    </Typography>
                                                    :
                                                    <Typography
                                                        variant="caption"
                                                        sx={{
                                                            color: theme.palette.text.disabled,
                                                            textDecoration: "line-through"
                                                        }}
                                                    >
                                                        {element?.discount}
                                                    </Typography>
                                                }
                                            </>
                                            :
                                            <Typography
                                                variant="body2"
                                                color={
                                                    i === 0 ?
                                                        theme.palette.text.primary
                                                        : theme.palette.text.disabled}
                                            >
                                                {element}
                                            </Typography>
                                        }
                                    </Grid>
                                </Grid>
                            </TableCell>
                        ))}
                    </TableRow>
                ))}
            </TableBody>
        </MUITable>
    );
};

export default Table;