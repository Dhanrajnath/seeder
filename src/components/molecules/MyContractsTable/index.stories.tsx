import { Meta, StoryObj } from "@storybook/react";
import MyContractsTable, { IMyContractsTableProps } from ".";
import { myContractsTableColumnLabels, myContractsTableData } from "../../../utils/myContracts";

export default {
    title: 'Molecules/MyContractsTable',
    component: MyContractsTable,
} as Meta;

type MyContractsTableType = StoryObj<IMyContractsTableProps>;


export const MyContracts: MyContractsTableType = {
    args: {
        tableColumns: myContractsTableColumnLabels,
        tableData: myContractsTableData,
    }
};
