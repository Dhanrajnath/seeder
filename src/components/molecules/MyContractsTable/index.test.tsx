import { render } from "@testing-library/react";
import MyContractsTable from ".";
import { myContractsTableColumnLabels, myContractsTableData } from "../../../utils/myContracts";

describe('MyContractsTable test', () => {
    it('renders the contracts table with data', () => {
        const { getByText, container } = render(
            <MyContractsTable
                tableColumns={myContractsTableColumnLabels}
                tableData={myContractsTableData}
            />
        );

        expect(getByText('Name')).toBeInTheDocument();
        expect(getByText('Type')).toBeInTheDocument();
        expect(getByText('Per payment')).toBeInTheDocument();
        expect(getByText('Term length')).toBeInTheDocument();
        expect(getByText('Payment amount')).toBeInTheDocument();

        expect(getByText('Contracts1')).toBeInTheDocument();
        expect(getByText('$12,000.25')).toBeInTheDocument();
        expect(getByText('$126,722.64')).toBeInTheDocument();

        expect(container.firstChild).toMatchSnapshot();
    });
});
