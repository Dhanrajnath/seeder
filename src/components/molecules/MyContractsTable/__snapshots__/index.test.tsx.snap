// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyContractsTable test renders the contracts table with data 1`] = `
<table
  class="MuiTable-root css-1ekzhym-MuiTable-root"
>
  <thead
    class="MuiTableHead-root css-r600lz-MuiTableHead-root"
  >
    <tr
      class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
    >
      <th
        class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-4d0fmp-MuiTableCell-root"
        scope="col"
        width="20%"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-ywzhw-MuiTypography-root"
        >
          Name
        </p>
      </th>
      <th
        class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-4d0fmp-MuiTableCell-root"
        scope="col"
        width="20%"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-ywzhw-MuiTypography-root"
        >
          Type
        </p>
      </th>
      <th
        class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-4d0fmp-MuiTableCell-root"
        scope="col"
        width="20%"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-ywzhw-MuiTypography-root"
        >
          Per payment
        </p>
      </th>
      <th
        class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-4d0fmp-MuiTableCell-root"
        scope="col"
        width="20%"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-ywzhw-MuiTypography-root"
        >
          Term length
        </p>
      </th>
      <th
        class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-4d0fmp-MuiTableCell-root"
        scope="col"
        width="20%"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-ywzhw-MuiTypography-root"
        >
          Payment amount
        </p>
      </th>
    </tr>
  </thead>
  <tbody
    class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
  >
    <tr
      class="MuiTableRow-root css-1xe9h37-MuiTableRow-root"
    >
      <td
        class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1kwzrlz-MuiTableCell-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-qfugio-MuiTypography-root"
            >
              Contracts1
            </p>
          </div>
        </div>
      </td>
      <td
        class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1kwzrlz-MuiTableCell-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-ywzhw-MuiTypography-root"
            >
              Monthly
            </p>
          </div>
        </div>
      </td>
      <td
        class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1kwzrlz-MuiTableCell-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-ywzhw-MuiTypography-root"
            >
              $12,000.25
            </p>
          </div>
        </div>
      </td>
      <td
        class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1kwzrlz-MuiTableCell-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-bptfh-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-ywzhw-MuiTypography-root"
            >
              12 months
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-192l7z3-MuiTypography-root"
            >
              12.0% fee
            </span>
          </div>
        </div>
      </td>
      <td
        class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1kwzrlz-MuiTableCell-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-bptfh-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-ywzhw-MuiTypography-root"
            >
              $126,722.64
            </p>
          </div>
        </div>
      </td>
    </tr>
    <tr
      class="MuiTableRow-root css-1xe9h37-MuiTableRow-root"
    >
      <td
        class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1kwzrlz-MuiTableCell-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-qfugio-MuiTypography-root"
            >
              Contracts2
            </p>
          </div>
        </div>
      </td>
      <td
        class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1kwzrlz-MuiTableCell-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-ywzhw-MuiTypography-root"
            >
              Monthly
            </p>
          </div>
        </div>
      </td>
      <td
        class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1kwzrlz-MuiTableCell-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-ywzhw-MuiTypography-root"
            >
              $6,000.00
            </p>
          </div>
        </div>
      </td>
      <td
        class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1kwzrlz-MuiTableCell-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-bptfh-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-ywzhw-MuiTypography-root"
            >
              12 months
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-192l7z3-MuiTypography-root"
            >
              12.0% fee
            </span>
          </div>
        </div>
      </td>
      <td
        class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1kwzrlz-MuiTableCell-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-bptfh-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-ywzhw-MuiTypography-root"
            >
              $23,277.36
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-vhzd1m-MuiTypography-root"
            >
              $63,380.00
            </span>
          </div>
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
